posthog/api/test/test_insight.py

```diff
@@ -3556,3 +3559,60 @@ def test_insight_returns_cached_types(self) -> None:
 
         self.assertNotIn("code", response)
         self.assertIsNotNone(response["results"][0]["types"])
+
+    def test_insight_variables_overrides(self):
+        dashboard = Dashboard.objects.create(
+            team=self.team,
+            name="dashboard 1",
+            created_by=self.user,
+        )
+        variable = InsightVariable.objects.create(
+            team=self.team, name="Test 1", code_name="test_1", default_value="some_default_value", type="String"
+        )
+        insight = Insight.objects.create(
+            filters={},
+            query={
+                "kind": "DataVisualizationNode",
+                "source": {
+                    "kind": "HogQLQuery",
+                    "query": "select {variables.test_1}",
```

Comment:
How does this affect saving views right now?

Reply:
This shouldn't be allowed to be saved as a view - it should error out due to it having placeholder nodes, but I can check this before the flag is switched